### ISSUE TITLE
Random failure in Unit.Evolution.Systems.ScalarWave.Characteristics 

### ISSUE THREAD

**Issue by nilsdeppe** [sxs-collaboration/spectre#2078]
# Bug reports:

```
2020-03-10T23:12:01.5935721Z 509/1106 Test #501: Unit.Evolution.Systems.ScalarWave.Characteristics ...........................***Exception: Child aborted  1.32 sec
2020-03-10T23:12:01.5936478Z Charm++: standalone mode (not using charmrun)
2020-03-10T23:12:01.5937432Z Charm++> Running in Multicore mode:  1 threads
2020-03-10T23:12:01.5939055Z Converse/Charm++ Commit ID: v6.8.0-0-ga36028edb
2020-03-10T23:12:01.5939859Z Warning> Randomization of virtual memory (ASLR) is turned on in the kernel, thread migration may not work! Run 'echo 0 > /proc/sys/kernel/randomize_va_space' as root to disable it, or try running with '+isomalloc_sync'.
2020-03-10T23:12:01.5940295Z CharmLB> Load balancer assumes all CPUs are same.
2020-03-10T23:12:01.5940808Z Charm++> Running on 1 unique compute nodes (2-way SMP).
2020-03-10T23:12:01.5941177Z Charm++> cpu topology info is gathered in 0.000 seconds.
2020-03-10T23:12:01.5941483Z SpECTRE Build Information:
2020-03-10T23:12:01.5941774Z Version:                      0.0.0
2020-03-10T23:12:01.5942065Z Compiled on host:             39812fb9f6f8
2020-03-10T23:12:01.5942343Z Compiled in directory:        /work/build
2020-03-10T23:12:01.5942641Z Source directory is:          /__w/holle/holle
2020-03-10T23:12:01.5942934Z Compiled on git branch:       moving_mesh_tags3
2020-03-10T23:12:01.5943321Z Compiled with git hash:       c255cf0b87a4f5c211983bdba9088f6b9bb98580
2020-03-10T23:12:01.5943630Z Linked on:                    Tue Mar 10 22:22:06 2020
2020-03-10T23:12:01.5943916Z Filters: "Unit.Evolution.Systems.ScalarWave.Characteristics"
2020-03-10T23:12:01.5944670Z 
2020-03-10T23:12:01.5945395Z ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2020-03-10T23:12:01.5945683Z "Release.Unit.Evolution.Systems.ScalarWave.Characteristics" is a Catch v2.11.1 host application.
2020-03-10T23:12:01.5946169Z Run with -? for options
2020-03-10T23:12:01.5946453Z 
2020-03-10T23:12:01.5946936Z -------------------------------------------------------------------------------
2020-03-10T23:12:01.5947254Z Unit.Evolution.Systems.ScalarWave.Characteristics
2020-03-10T23:12:01.5947735Z -------------------------------------------------------------------------------
2020-03-10T23:12:01.5948087Z /__w/holle/holle/tests/Unit/Evolution/Systems/ScalarWave/Test_Characteristics.cpp:336
2020-03-10T23:12:01.5948389Z ...............................................................................
2020-03-10T23:12:01.5948609Z 
2020-03-10T23:12:01.5948876Z /__w/holle/holle/tests/Unit/TestingFramework.hpp:162: FAILED:
2020-03-10T23:12:01.5949146Z   CHECK( a == appx(b) )
2020-03-10T23:12:01.5949414Z with expansion:
2020-03-10T23:12:01.5949851Z   -1.5557258474 == Approx( -1.5557258474 )
2020-03-10T23:12:01.5950185Z with messages:
2020-03-10T23:12:01.5950459Z   Seed is: 1843061027 from /__w/holle/holle/tests/Unit/Pypp/
2020-03-10T23:12:01.5950729Z   CheckWithRandomValues.hpp:356
2020-03-10T23:12:01.5950994Z   function: char_field_vzero
2020-03-10T23:12:01.5951265Z   /__w/holle/holle/tests/Unit/Pypp/CheckWithRandomValues.hpp:230: result ==
2020-03-10T23:12:01.5951846Z   pypp::call<ResultType>( module_name, function_name, std::get<ArgumentIs>
2020-03-10T23:12:01.5952228Z   (args)..., forward_to_pypp<ResultType>(std::get<MemberArgsIs>(member_args),
2020-03-10T23:12:01.5952533Z   used_for_size)...)
2020-03-10T23:12:01.5953047Z   a: T(0)=(2.987892498956078198e+05,-6.984108854495642299e+04,-1.
2020-03-10T23:12:01.5953599Z   357962548813461581e+02,-4.577756722409256327e+04,5.073271355190077447e+05)
2020-03-10T23:12:01.5954134Z   T(1)=(7.717785799971950473e+04,-6.465411525544807955e+04,-1.
2020-03-10T23:12:01.5954935Z   555725847403401474e+00,2.327561438303340401e+04,-1.138555726179508747e+04)
2020-03-10T23:12:01.5955475Z   T(2)=(-9.539125931104605843e+04,-8.333352608110170695e+04,7.
2020-03-10T23:12:01.5956192Z   269392619228574404e+01,-2.396001273430423225e+03,-2.755399576082115527e+04)
2020-03-10T23:12:01.5956717Z   b: T(0)=(2.987892498956078198e+05,-6.984108854495643754e+04,-1.
2020-03-10T23:12:01.5957274Z   357962548814248294e+02,-4.577756722409254871e+04,5.073271355190077447e+05)
2020-03-10T23:12:01.5957974Z   T(1)=(7.717785799971950473e+04,-6.465411525544807955e+04,-1.
2020-03-10T23:12:01.5958668Z   555725847370922565e+00,2.327561438303341856e+04,-1.138555726179508929e+04)
2020-03-10T23:12:01.5959909Z   T(2)=(-9.539125931104604388e+04,-8.333352608110175061e+04,7.
2020-03-10T23:12:01.5960524Z   269392619229620323e+01,-2.396001273430423680e+03,-2.755399576082115891e+04)
2020-03-10T23:12:01.5960955Z   a: (7.717785799971950473e+04,-6.465411525544807955e+04,-1.
2020-03-10T23:12:01.5961388Z   555725847403401474e+00,2.327561438303340401e+04,-1.138555726179508747e+04)
2020-03-10T23:12:01.5962235Z   b: (7.717785799971950473e+04,-6.465411525544807955e+04,-1.
2020-03-10T23:12:01.5962699Z   555725847370922565e+00,2.327561438303341856e+04,-1.138555726179508929e+04)
2020-03-10T23:12:01.5963100Z   a: -1.555725847403401474e+00
2020-03-10T23:12:01.5963504Z   b: -1.555725847370922565e+00
2020-03-10T23:12:01.5963705Z 
2020-03-10T23:12:01.5963906Z 0.384 s: Unit.Evolution.Systems.ScalarWave.Characteristics
2020-03-10T23:12:01.5964131Z ===============================================================================
2020-03-10T23:12:01.5964356Z test cases:     1 |     0 passed | 1 failed
2020-03-10T23:12:01.5964581Z assertions: 21213 | 21212 passed | 1 failed
2020-03-10T23:12:01.5964742Z 
```

**Comment by nilsdeppe** [comment]
And another...

```
2020-03-11T18:27:36.2743343Z with expansion:
2020-03-11T18:27:36.2743462Z   14.7699883299 == Approx( 14.7699883298 )
2020-03-11T18:27:36.2743583Z with messages:
2020-03-11T18:27:36.2743709Z   Seed is: 3588409903 from /__w/spectre/spectre/tests/Unit/Pypp/
2020-03-11T18:27:36.2743840Z   CheckWithRandomValues.hpp:356
2020-03-11T18:27:36.2743961Z   function: char_field_vzero
2020-03-11T18:27:36.2744075Z   /__w/spectre/spectre/tests/Unit/Pypp/CheckWithRandomValues.hpp:230: result ==
2020-03-11T18:27:36.2744480Z   pypp::call<ResultType>( module_name, function_name, std::get<ArgumentIs>
2020-03-11T18:27:36.2744666Z   (args)..., forward_to_pypp<ResultType>(std::get<MemberArgsIs>(member_args),
2020-03-11T18:27:36.2744800Z   used_for_size)...)
2020-03-11T18:27:36.2745146Z   a: T(0)=(1.476998832988483912e+01,-5.911233634252906777e+05,2.
2020-03-11T18:27:36.2745292Z   246473283611016523e+05,1.415170040798651462e+05,2.889131867293043761e+05)
2020-03-11T18:27:36.2745620Z   T(1)=(-5.781860876062836496e+00,3.512291002372363582e+05,1.
2020-03-11T18:27:36.2745952Z   334391574894866790e+05,-3.271115115981466370e+05,3.596409463445380097e+05)
2020-03-11T18:27:36.2746282Z   T(2)=(1.925283087219850984e+02,3.003121318052576971e+05,-2.
2020-03-11T18:27:36.2746417Z   303604645477711529e+04,2.985769370406732196e+05,1.190738558589717431e+05)
2020-03-11T18:27:36.2746744Z   b: T(0)=(1.476998832979006693e+01,-5.911233634252906777e+05,2.
2020-03-11T18:27:36.2746880Z   246473283611015941e+05,1.415170040798651171e+05,2.889131867293043761e+05)
2020-03-11T18:27:36.2747208Z   T(1)=(-5.781860876129940152e+00,3.512291002372363582e+05,1.
2020-03-11T18:27:36.2747545Z   334391574894866499e+05,-3.271115115981465788e+05,3.596409463445379515e+05)
2020-03-11T18:27:36.2747853Z   T(2)=(1.925283087219577283e+02,3.003121318052576389e+05,-2.
2020-03-11T18:27:36.2747984Z   303604645477711165e+04,2.985769370406732196e+05,1.190738558589717722e+05)
2020-03-11T18:27:36.2748335Z   a: (1.476998832988483912e+01,-5.911233634252906777e+05,2.246473283611016523e+
2020-03-11T18:27:36.2748470Z   05,1.415170040798651462e+05,2.889131867293043761e+05)
2020-03-11T18:27:36.2748810Z   b: (1.476998832979006693e+01,-5.911233634252906777e+05,2.246473283611015941e+
2020-03-11T18:27:36.2748943Z   05,1.415170040798651171e+05,2.889131867293043761e+05)
2020-03-11T18:27:36.2749066Z   a: 1.476998832988483912e+01
2020-03-11T18:27:36.2749170Z   b: 1.476998832979006693e+01
```

**Comment by nilsdeppe** [comment]
```
2020-03-26T03:58:53.7257170Z /__w/spectre/spectre/tests/Unit/Evolution/Systems/ScalarWave/Test_Characteristics.cpp:336
2020-03-26T03:58:53.7257307Z ...............................................................................
2020-03-26T03:58:53.7257368Z 
2020-03-26T03:58:53.7257597Z /__w/spectre/spectre/tests/Unit/Framework/TestingFramework.hpp:166: FAILED:
2020-03-26T03:58:53.7258375Z   CHECK( a == appx(b) )
2020-03-26T03:58:53.7258511Z with expansion:
2020-03-26T03:58:53.7258876Z   -0.9036941909 == Approx( -0.9036941909 )
2020-03-26T03:58:53.7259014Z with messages:
2020-03-26T03:58:53.7259151Z   Seed is: 3540729235 from /__w/spectre/spectre/tests/Unit/Framework/
2020-03-26T03:58:53.7259299Z   CheckWithRandomValues.hpp:359
2020-03-26T03:58:53.7259433Z   function: char_field_vzero
2020-03-26T03:58:53.7259577Z   /__w/spectre/spectre/tests/Unit/Framework/CheckWithRandomValues.hpp:233:
2020-03-26T03:58:53.7266598Z   result == pypp::call<ResultType>( module_name, function_name, std::get
2020-03-26T03:58:53.7266957Z   <ArgumentIs>(args)..., forward_to_pypp<ResultType>(std::get<MemberArgsIs>
2020-03-26T03:58:53.7267093Z   (member_args), used_for_size)...)
2020-03-26T03:58:53.7267215Z   a: T(0)=(2.345713904417619779e+04,9.950501417194482201e+03,3.
2020-03-26T03:58:53.7267664Z   520368286052717594e+05,-9.036941908710787175e-01,-5.147327381610947195e+05)
2020-03-26T03:58:53.7268206Z   T(1)=(-8.311323196947255637e+05,-1.004093098315301359e+04,-3.
2020-03-26T03:58:53.7268567Z   227561044805530692e+05,-1.271761474645057035e+02,5.833815401018615376e+03)
2020-03-26T03:58:53.7268850Z   T(2)=(-6.310992102823782479e+05,3.373407191456156852e+03,1.
2020-03-26T03:58:53.7269144Z   332740767181141418e+05,1.955249816976815112e+02,-3.494953874819110497e+05)
2020-03-26T03:58:53.7269276Z   b: T(0)=(2.345713904417619779e+04,9.950501417194485839e+03,3.
2020-03-26T03:58:53.7269763Z   520368286052717594e+05,-9.036941908634616993e-01,-5.147327381610948360e+05)
2020-03-26T03:58:53.7270399Z   T(1)=(-8.311323196947255637e+05,-1.004093098315300449e+04,-3.
2020-03-26T03:58:53.7270894Z   227561044805531274e+05,-1.271761474644299597e+02,5.833815401018615376e+03)
2020-03-26T03:58:53.7271392Z   T(2)=(-6.310992102823782479e+05,3.373407191456155488e+03,1.
2020-03-26T03:58:53.7271714Z   332740767181141418e+05,1.955249816976138391e+02,-3.494953874819111079e+05)
2020-03-26T03:58:53.7271878Z   a: (2.345713904417619779e+04,9.950501417194482201e+03,3.520368286052717594e+
2020-03-26T03:58:53.7272331Z   05,-9.036941908710787175e-01,-5.147327381610947195e+05)
2020-03-26T03:58:53.7272466Z   b: (2.345713904417619779e+04,9.950501417194485839e+03,3.520368286052717594e+
2020-03-26T03:58:53.7272927Z   05,-9.036941908634616993e-01,-5.147327381610948360e+05)
2020-03-26T03:58:53.7273521Z   a: -9.036941908710787175e-01
2020-03-26T03:58:53.7273981Z   b: -9.036941908634616993e-01
2020-03-26T03:58:53.7274049Z 
```

**Comment by geoffrey4444** [comment]
Reproduced failure with seed 2478544957. Loosening tolerance from 1e-12 to 1e-11 led to the test passing again. 

PR to implement this fix: #2176
